src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordPrintMembers.cs

```diff
@@ -0,0 +1,277 @@
+                if (ContainingType.BaseTypeNoUseSiteDiagnostics.IsObjectType())
+                {
+                    if (printableMembers.IsEmpty)
+                    {
+                        // return false;
+                        F.CloseMethod(F.Return(F.Literal(false)));
+                        return;
+                    }
+                }
+                else
+                {
+                    MethodSymbol? printMethod = FindValidPrintMethod(ContainingType.BaseTypeNoUseSiteDiagnostics, DeclaringCompilation);
+                    if (printMethod is null)
+                    {
+                        F.CloseMethod(F.ThrowNull()); // an error was reported in base checks already
+                        return;
+                    }
+
+                    var basePrintCall = F.Call(receiver: F.Base(ContainingType.BaseTypeNoUseSiteDiagnostics), printMethod, builder);
+                    if (printableMembers.IsEmpty)
+                    {
+                        // return base.print(builder);
+                        F.CloseMethod(F.Return(basePrintCall));
+                        return;
+                    }
+                    else
+                    {
+                        // if (base.print(builder))
+                        //     builder.Append(", ")
+                        block!.Add(
+                            F.If(basePrintCall,
+                                F.ExpressionStatement(F.Call(receiver: builder, F.WellKnownMethod(WellKnownMember.System_StringBuilder__AppendString), F.StringLiteral(", ")))));
+                    }
+                }
+
+                Debug.Assert(!printableMembers.IsEmpty && block is object);
+
+                for (var i = 0; i < printableMembers.Length; i++)
+                {
+                    // builder.Append(<name>);
+                    // builder.Append(" = ");
+                    // builder.Append((object)<value>);
+                    // builder.Append(", "); // except for last member
+
+                    var member = printableMembers[i];
+                    block.Add(
+                        F.ExpressionStatement(F.Call(receiver: builder,
+                            F.WellKnownMethod(WellKnownMember.System_StringBuilder__AppendString), F.StringLiteral(member.Name))));
+                    block.Add(
+                        F.ExpressionStatement(F.Call(receiver: builder,
+                            F.WellKnownMethod(WellKnownMember.System_StringBuilder__AppendString), F.StringLiteral(" = "))));
+
+                    var value = member.Kind switch
+                    {
+                        SymbolKind.Field => F.Field(F.This(), (FieldSymbol)member),
+                        SymbolKind.Property => F.Property(F.This(), (PropertySymbol)member),
+                        _ => throw ExceptionUtilities.UnexpectedValue(member.Kind)
+                    };
+
+                    block.Add(F.ExpressionStatement(
```

Comment:
sorry, I might have missed some context in the proposal PR, but doesn't it make the most sense to:

1. convert to object if member is a reference type and call StringBuilder.Append(object)
2. call ToString() if member is a value type and call StringBuilder.Append(string)

Reply:
I'll take a note (I think it's a good idea to reduce boxing), but that's not the spec behavior that we discussed so far.
[https://github.com/dotnet/csharplang/blob/master/proposals/csharp-9.0/records.md#printing-members-printmembers-and-tostring-methods](https://github.com/dotnet/csharplang/blob/master/proposals/csharp-9.0/records.md#printing-members-printmembers-and-tostring-methods)

---
In reply to: [469410413](https://github.com/dotnet/roslyn/pull/46616#discussion_r469410413) [](ancestors = 469410413)

Reply:
Added note to LDM section in test plan: [https://github.com/dotnet/roslyn/issues/40726](https://github.com/dotnet/roslyn/issues/40726)

---
In reply to: [469461541](https://github.com/dotnet/roslyn/pull/46616#discussion_r469461541) [](ancestors = 469461541,469410413)